### PR TITLE
Add support for the Astra DB Reader

### PR DESCRIPTION
**Pull request by erichare**
# Description

This adds support for a reader class for Astra DB following the conventions of the repository.

## Type of Change

Please delete options that are not relevant.

- [x] New Loader/Tool
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] I stared at the code and made sure it makes sense
- [x] I added an end-to-end README file

# Suggested Checklist:

- [x] I have added a library.json file if a new loader/tool was added
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: tests/tests_snowflake/test_snowflake_base.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 from llama_hub.snowflake.base import SnowflakeReader
 from llama_index.readers.schema.base import Document
 
+
```
Comment (nerdai) [inline tests/tests_snowflake/test_snowflake_base.py:6]: also this? or was this a result from lint/fmt?

**Review by nerdai — APPROVED** [review]
This looks great @erichare -- thanks for the contribution! :) 

Looks like we just need to get those unit tests to pass.

File: tests/tests_snowflake/test_snowflake_base.py (hunk 1 above)
Comment (erichare) [reply tests/tests_snowflake/test_snowflake_base.py:6]: ^ yep, this was also from the linter - i can remove from this PR if you prefer.

Comment (nerdai) [reply tests/tests_snowflake/test_snowflake_base.py:6]: all good, let's just leave them both thanks!

**Comment by nerdai** [comment]
> > This looks great @erichare -- thanks for the contribution! :)
> > Looks like we just need to get those unit tests to pass.
> 
> I wanted to ask, i noticed them failing. I thought i added a valid id to the library.json, but it seems like the tests are saying its missing? I'll look more closely once i wake up a bit more, but if you have an idea what i may have missed im all ears. Thank you so much for reviewing!

Oh okay, yea let me take a look real quick and see if I can figure out what's happening here!

File: llama_hub/library.json [hunk 2]
```diff
@@ -39,6 +39,10 @@
     "id": "assemblyai",
     "author": "patrickloeber"
   },
+  "AstraDBReader": {
+    "id": "astra_db_reader",
```
Comment (nerdai) [inline llama_hub/library.json:43]: I think i found the reason why tests are failing: `id` here should be the same as the directory name which in this case is `astra_db`

Comment (erichare) [reply llama_hub/library.json:43]: Aha, my mistake!! yes i see that now. I'll push the update shortly. thank you!

**Comment by erichare** [comment]
@nerdai looks like that fixed it :)

**Comment by nerdai** [comment]
> @nerdai looks like that fixed it :)

🎉 woohoo! thanks again @erichare  -- going to merge this now.

**Pull request merged**